contracts/farms/RevoMobiusFarmBot.sol

```diff
@@ -0,0 +1,209 @@
+        );
+        celoNativeStakingTokenIndex = _celoNativeStakingTokenIndex;
+        bridgedStakingTokenIndex = 1 - celoNativeStakingTokenIndex;
+
+        if (celoNativeStakingTokenIndex == 0) {
+            celoNativeStakingToken = stakingToken0;
+        } else {
+            celoNativeStakingToken = stakingToken1;
+        }
+    }
+
+    function _deposit(uint256 _lpAmount) internal override whenNotPaused {
+        require(_lpAmount > 0, "Cannot invest in farm because lpAmount is 0");
+        stakingToken.safeIncreaseAllowance(address(liquidityGauge), _lpAmount);
+        liquidityGauge.deposit(_lpAmount);
+    }
+
+    function _withdraw(uint256 _lpAmount) internal override {
+        liquidityGauge.withdraw(_lpAmount);
+    }
+
+    function _claimRewards() internal whenNotPaused {
+        minter.mint(address(liquidityGauge));
+        liquidityGauge.claim_rewards(address(this));
+    }
+
+    function compound(
+        address[][] memory _paths,
+        uint256[] memory _minAmountsOut,
+        uint256 _minSwapOut,
+        uint256 _minLiquidity,
+        uint256 _deadline
+    ) external ensure(_deadline) onlyRole(COMPOUNDER_ROLE) whenNotPaused {
+        // Important safety checks
+        require(
+            _paths.length == rewardsTokens.length,
+            "Parameter _paths must have length equal to rewardsTokens"
+        );
+        require(
+            _minAmountsOut.length == rewardsTokens.length,
+            "Parameter _minAmountsOut must have length equal to rewardsTokens"
+        );
+        for (uint256 i = 0; i < _paths.length; i++) {
+            require(
+                _paths[i][_paths[i].length - 1] ==
+                    address(celoNativeStakingToken),
+                "Each swap path must end with the Celo-native staking token"
+            );
+        }
+
+        // Get native and external rewards
+        _claimRewards();
+
+        // Get the current balance of rewards tokens
+        uint256[] memory _tokenBalances = new uint256[](rewardsTokens.length);
+        for (uint256 i = 0; i < rewardsTokens.length; i++) {
+            _tokenBalances[i] = rewardsTokens[i].balanceOf(address(this));
+        }
+
+        // Swap rewards for the Celo-native LP constituent token
+        for (uint256 i = 0; i < rewardsTokens.length; i++) {
+            UniswapRouter.swap(
+                router,
+                _paths[i],
+                _tokenBalances[i],
+                rewardsTokens[i],
+                _minAmountsOut[i],
+                _deadline
+            );
+        }
+
+        uint256 halfCeloNativeStakingTokenBalance = celoNativeStakingToken
+            .balanceOf(address(this)) / 2;
+
+        // Take half the Celo-native staking token and swap it for the bridged version
+        celoNativeStakingToken.safeIncreaseAllowance(
+            address(swap),
+            halfCeloNativeStakingTokenBalance
+        );
+
+        swap.swap(
```

Comment:
During testing, I noticed some very strange behavior with this that led me down a rabbit hole. This was always swapping for 0 of the bridged asset. It turns out, bridged assets' ERC20 implementations have a different number of decimals than is typical (i.e., not 18). This means that for small enough amounts of the native token, there's _no amount_ of the bridged token that exists that's small enough to match the value of the native token. e.g. if 100 cUSD corresponds to 1 cUSDC, then 20 cUSD is most closely equal to 0 cUSDC.

All this to say I haven't been able to test this swap yet, since I'm testing with very small amounts of LP so the rewards are very small.

Reply:
This in turn blocks the `addLiquidity` call later. Internally, `addLiquidity` checks to make sure that all the provided LP amounts are non-zero.

Reply:
no longer relevant!